Emby.Server.Implementations/AppBase/BaseApplicationPaths.cs

```diff
@@ -14,48 +15,46 @@ public abstract class BaseApplicationPaths : IApplicationPaths
         /// </summary>
         protected BaseApplicationPaths(
             string programDataPath,
-            string appFolderPath,
-            string logDirectoryPath = null,
-            string configurationDirectoryPath = null)
+            string logDirectoryPath,
+            string configurationDirectoryPath)
         {
-            ProgramDataPath = programDataPath;
-            ProgramSystemPath = appFolderPath;
-            LogDirectoryPath = logDirectoryPath;
-            ConfigurationDirectoryPath = configurationDirectoryPath;
+            try
```

Comment:
This logic shouldn't be here at all.

Reply:
Better than in the getters, right?

Reply:
You can create a function "EnsureCreated" or something along those lines.

Reply:
Still to much logic in this constructor.

Reply:
I'm suggesting that logic comes out the constructor and moves to new method in Programs.cs, called EnsureCreated().  ?